controllers/storagecluster/rookoperatorhostnetwork.go

```diff
@@ -0,0 +1,104 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+
+	opv1a1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type rookCephCsvHostNetwork struct{}
+
+// if the StorageCluster is configured to run on non-default host network, rook-ceph-operator needs to run on host network as well
+// this is validated by checking if the AddressRanges.Public field is set in the Network spec
+// since pod network cannot always communicate with non-default host network rook-ceph-operator needs to run on host network
+func (obj *rookCephCsvHostNetwork) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	if shouldUseHostNetworking(instance) {
+		err := updateOpHostNetworkInRookCephCSV(r.ctx, r.Client, instance.Namespace, true)
+		if err != nil {
+			r.Log.Error(err, "Failed to update Rook Ceph Operator HostNetwork.", "StorageCluster", klog.KRef(instance.Namespace, instance.Name))
+			return reconcile.Result{}, err
+		}
+	}
+	return reconcile.Result{}, nil
+}
```

Comment:
What is going to happen in mixed mode, when we have more then one storagecluster/cephcluster?

Reply:
If there is another storageCluster which is not using dedicated hostNetwork will break the current scenario.
Is it possible to restrict only one storage cluster creation, or we set the hostNetwork for rook-op once but other storageCluster may error out if not using host-net